logstash-core/lib/logstash/pipeline.rb

```diff
@@ -82,14 +82,13 @@ def initialize(config_str, settings = LogStash::SETTINGS, namespaced_metric = ni
       raise
     end
 
-    @input_queue = LogStash::Util::WrappedSynchronousQueue.new
+    queue = LogStash::Util::WrappedSynchronousQueue.new
+    @input_queue_client = queue.write_client
+    @filter_queue_client = queue.read_client
```

Comment:
The client is an adapter that presents one API to the each side of any given queue implementation. There also may be different concerns regarding batching and thread-safety guarantees between read and write that may complicate a dual purpose client. Further, separate clients guarantee that an input can't read an event from the queue and a worker can't write an event to the queue - which suits our current requirements.

The DLQ client, for example, should only be able to write events out.

It is possible to have per input client instances and each instance may post metrics on that input's interaction with the queue without the input needing to know or do this.

But given that a batch is isolated to a worker thread then that thread's interaction with the queue to retrieve a batch can post metrics on a per thread per batch basis.

In this PR the batch metrics are thread anonymous - that can be a future enhancement.